uwg/RSMDef.py

```diff
@@ -74,314 +105,323 @@ def __init__(self,lat,lon,GMT,height,T_init,P_init,parameter,z_meso_path):
-            self.tempProf[0], rural.sens, self.nzref, forc.wind, \
-            self.tempProf, parameter)
-
+        cd, ustarRur = self.DiffusionCoefficient(
+            self.densityProfC[0], self.z, self.dz, self.z0r, self.disp, self.tempProf[0],
+            rural.sens, self.nzref, forc.wind, self.tempProf, parameter)
 
         # solve diffusion equation
-        self.tempProf = self.DiffusionEquation(self.nzref,simTime.dt,\
-            self.tempProf,self.densityProfC,self.densityProfS,cd,self.dz)
+        self.tempProf = RSMDef.DiffusionEquation(
+            self.nzref, simTime.dt, self.tempProf, self.densityProfC, self.densityProfS,
+            cd, self.dz)
 
         # compute wind profile
         # N.B In Matlab, negative values are converted to complex values.
         # log(-x) = log(x) + log(-1) = log(x) + i*pi
         # Python will throw an exception. Negative value occurs here if
         # VDM is run for average obstacle height ~ 4m.
         for iz in range(self.nzref):
-            self.windProf[iz] = ustarRur/parameter.vk*\
-                math.log((self.z[iz]-self.disp)/self.z0r)
+            self.windProf[iz] = \
+                ustarRur / parameter.vk * log((self.z[iz] - self.disp) / self.z0r)
 
         # Average pressure
         self.ublPres = 0.
         for iz in range(self.nzfor):
-            self.ublPres = self.ublPres + \
-                self.presProf[iz]*self.dz[iz]/(self.z[self.nzref-1]+self.dz[self.nzref-1]/2.)
-
-    def DiffusionEquation(self,nz,dt,co,da,daz,cd,dz):
-
-        cddz = [0 for i in range(nz+2)]
-        a = [[0 for j in range(3)] for i in range(nz)]
-        c = [0 for i in range(nz)]
-
-        #--------------------------------------------------------------------------
-        cddz[0] = daz[0]*cd[0]/dz[0]
-        for iz in range(1,nz):
-            cddz[iz] = 2.*daz[iz]*cd[iz]/(dz[iz]+dz[iz-1])
-        cddz[nz] = daz[nz]*cd[nz]/dz[nz]
-        #--------------------------------------------------------------------------
-        a[0][0] = 0.
-        a[0][1] = 1.
-        a[0][2] = 0.
-        c[0] = co[0]
-
-        for iz in range(1,nz-1):
-            dzv = dz[iz]
-            a[iz][0]=-cddz[iz]*dt/dzv/da[iz]
-            a[iz][1]=1+dt*(cddz[iz]+cddz[iz+1])/dzv/da[iz]
-            a[iz][2]=-cddz[iz+1]*dt/dzv/da[iz]
-            c[iz]=co[iz]
-
-        a[nz-1][0]=-1.
-        a[nz-1][1]=1.
-        a[nz-1][2]=0.
-        c[nz-1]=0.
+            self.ublPres = \
+                self.ublPres + self.presProf[iz] * self.dz[iz] / \
+                (self.z[self.nzref-1] + self.dz[self.nzref-1] / 2.)
 
-        #--------------------------------------------------------------------------
-        co = self.invert(nz,a,c)
-        return co
-
-    def DiffusionCoefficient(self,rho,z,dz,z0,disp,tempRur,heatRur,nz,uref,th,parameter):
+    def DiffusionCoefficient(self, rho, z, dz, z0, disp, tempRur, heatRur, nz, uref, th,
+                             parameter):
```

Comment:
@chriswmackey 

I can change the capitalization (and I'll hunt down other methods where function capitilization occurs), but adding docstrings are a bit of a challenge. 

This is the one module in UWG that I have very little understanding of because it's relying on an understanding of physics that's beyond me (mesoscale fluid dynamics). Additionally, the original Matlab code uses nondescriptive, single-letter variable naming for almost everything, which is really confusing. I'd have to dig really deep into Bruno's thesis to be able to define docstrings. It is okay to keep it as is for now?

Reply:
Yes, don't worry about it if you don't actually know what the variables are.  Bruno should have been the one to document it and, now that he didn't, tracking down what it means is a whole other big task unto itself.  For now, it might help to put a `# TODO:` here that at least let's other folk know that writing the docstring requires us to do some mesoscale fluid dynamic research.